### PR TITLE
push a docker image to a registry

### PR DESCRIPTION
**Pull request by leo-ri**
- added a step to release workflow (push to the quay)
- k8s-demo-broker is pushing to quay too (or whatever in secret)
- renamed secrets/updated docs (registry related)
- .gitignore + .idea
- deleted test OSB version 1 (there are was only deploy broker)

### CODE REVIEW

File: .github/workflows/k8s-demo-broker.yml [hunk 1]
```diff
@@ -16,20 +16,29 @@ jobs:
...
       - name: Check out the repo
         uses: actions/checkout@v2
-      - name: Push Atlas-osb to Docker Hub
+      - name: Get Tag for the image
+        id: tag
+        run: |
+          branch=$(echo $GITHUB_REF | awk -F "/" '{print $3}')
+          echo "::set-output name=tag::demo-$branch"
+
+      - name: Push Atlas-osb to Registry
```
Comment (antonlisovenko) [inline .github/workflows/k8s-demo-broker.yml:25]: why does this action perform push as well? My understanding is that it's only the release process that is supposed to do this?

File: .github/workflows/k8s-demo-broker.yml [hunk 2]
```diff
@@ -45,7 +54,7 @@ jobs:
           echo "leori/atlas-osb:${branch_name}"
           helm install "${K_BROKER}" \
               --set namespace="${K_NAMESPACE}" \
-              --set image="leori/atlas-osb:${branch_name}" \
+              --set image="${INPUT_IMAGE}" \
```
Comment (antonlisovenko) [inline .github/workflows/k8s-demo-broker.yml:57]: [q] where is this variable defined?

File: .github/workflows/create-release-package.yml [hunk 3]
```diff
@@ -78,7 +78,16 @@ jobs:
...
         asset_content_type: application/tgz
+        
+    - name: Push Atlas-osb to Registry
+      uses: docker/build-push-action@v1
+      with:
+        username: ${{ secrets.REGISTRY_USERNAME }}
+        password: ${{ secrets.REGISTRY_PASSWORD }}
+        registry: ${{ secrets.REGISTRY }}
+        repository: ${{ secrets.REGISTRY_REPO }}
+        tags: ${{ steps.semver.outputs.version }}, latest
```
Comment (antonlisovenko) [inline .github/workflows/create-release-package.yml:93]: [q] where is this variable defined?

**Review by antonlisovenko — COMMENTED** [review]
Looks good.
The only important question is about duplication of push tag functionality in actions

File: .github/workflows/k8s-demo-broker.yml (hunk 2 above)
Comment (leo-ri) [reply .github/workflows/k8s-demo-broker.yml:57]: line 41 image argument

File: .github/workflows/k8s-demo-broker.yml (hunk 1 above)
Comment (leo-ri) [reply .github/workflows/k8s-demo-broker.yml:24]: now it is a local registry

File: .github/workflows/create-release-package.yml (hunk 3 above)
Comment (leo-ri) [reply .github/workflows/create-release-package.yml:93]: from action output https://github.com/mongodb/atlas-osb/blob/60647aca02adb8034e2fb65950d5bb4142465603/.github/workflows/create-release-package.yml#L45
https://github.com/leo-ri/increment-semver/blob/b43acb48b88f31fe1d5f7579baeada2ec9cdc58b/entrypoint.sh#L87

**Review by antonlisovenko — APPROVED** [review]
Looks good! 👍

**Pull request merged**